doc/source/serve/llm/user-guides/model-loading.md

```diff
@@ -249,7 +249,143 @@ Use EC2 instance profiles or EKS service accounts with appropriate S3 read permi
+    ...
+)
+```
+
+### Torch Compile Cache
+Torch.compile is a critical stage during setting up a model for inference, and it incurs some latency during initialization. This can be mitigated by keeping a torch compile cache, which is automatically generated by vLLM. To retrieve the torch compile cache, run vLLM and look for a log like below:
+```
+(RayWorkerWrapper pid=126782) INFO 10-15 11:57:04 [backends.py:608] Using cache directory: /home/ray/.cache/vllm/torch_compile_cache/131ee5c6d9/rank_1_0/backbone for vLLM's torch.compile
+```
+
+In this example the cache folder is located at `/home/ray/.cache/vllm/torch_compile_cache/131ee5c6d9`.
+Simply upload this file to your S3 bucket. The cache folder can now be retrieved at startup. We provide a custom utility to download the compile cache from cloud storage. Simply specify the `CloudDownloader` callback in `LLMConfig` and supply the relevant arguments. Make sure to set the `cache_dir` in compilation_config correctly. 
+
+```python
+llm_config = LLMConfig(
+    ...
+    callback_config={
+        "callback_class": "ray.llm._internal.common.callbacks.cloud_downloader.CloudDownloader",
+        "callback_kwargs": {"paths": [("s3://samplebucket/llama-3-8b-cache", "/home/ray/.cache/vllm/torch_compile_cache/llama-3-8b-cache")]},
+    },
+    engine_kwargs={
+        "tensor_parallel_size": 1,
+        "compilation_config": {
+            "cache_dir": "/home/ray/.cache/vllm/torch_compile_cache/llama-3-8b-cache",
+        }
+    },
+    ...
+)
+```
+
+### Benchmarked Results
+
+| LLama 8B (TP=1) | Model Load Time | Torch.compile Time | TTFT |
+| :---- | :---- | :---- | :---- |
+| Baseline (no optimizations) | 19.31 | 26.13 | 104.22 |
+| S3/RunAI Streamer | 4.07 | 26.27 | 86.42 |
+| Model Sharding | N/A | N/A | N/A |
+| Torch.compile Cache | 19.01 | 6.42 \+ (1.19 download) | 89.60 |
+| All Optimizations | 4.13 | 6.63 \+ (1.19 download) | 71.21 |
+
+
+| LLama 70B (TP=4) | Model Load Time | Torch.compile Time | TTFT |
+| :---- | :---- | :---- | :---- |
+| Baseline (no optimizations) | 213.54 | 60.79 | 359.55 |
+| S3/RunAI Streamer | 91.88 | 61.94 | 240.83 |
+| Model Sharding | 24.98 | 60.59 | 181.38 |
+| Torch.compile Cache | 233.08 | 15.49 \+ (4.58 download) | 338.39 |
+| All Optimizations | 27.51 | 15.65 \+ (4.58 download) | 125.98 |
+
+
+| Qwen 235B (TP=8) | Model Load Time | Torch.compile Time | TTFT |
+| :---- | :---- | :---- | :---- |
+| Baseline (no optimizations) | 756.96 | 111.09 | 994.71 |
+| S3/RunAI Streamer | 573.41 | 112.47 | 820.72 |
+| Model Sharding | 75.39 | 113.22 | 322.60 |
+| Torch.compile Cache | 715.11 | 22.71 \+ (14.79 download) | 892.17 |
+| All Optimizations | 75.84 | 22.65 \+ (14.79 download) | 256.54 |
+
+
+![Benchmark Results Visualization](../images/all_models_startup.png)
+![Benchmark Results Visualization](../images/llama8b_startup.png)
+![Benchmark Results Visualization](../images/llama70b_startup.png)
+![Benchmark Results Visualization](../images/qwen235b_startup.png)
+
```

Comment:
Add a bit of explaination on the interpretation from the graphs. i.e. 

baseline gets incrementally better as you use model loader + sharded setting + torch compile caching. cuda graph, fixed startup remain the most important candidates that can further be optimized in the engine and the community is looking at it (add github issue here)